eclipse-collections-code-generator/src/main/resources/api/map/objectPrimitiveMap.stg

```diff
@@ -94,6 +94,31 @@ public interface Object<name>Map\<K> extends <name>Iterable
     void forEachKeyValue(Object<name>Procedure\<? super K> procedure);
     <if(!primitive.booleanPrimitive)><(flipUniqueValues.(name))(name)><endif>
 
+    /**
+     * This method can be used to implement anySatisfy, allSatisfy, and noneSatisfy.
+     *
+     * anySatisfy  = this.shortCircuitKeyValue(predicate, true, true, false);
+     * allSatisfy  = this.shortCircuitKeyValue(predicate, false, false, true);
+     * noneSatisfy = this.shortCircuitKeyValue(predicate, true, false, true);
+     *
+     * @since 12.0
+     */
+    default boolean shortCircuitKeyValue(
+            Object<name>Predicate\<K> predicate,
+            boolean shortCircuitCondition,
+            boolean shortCircuitReturn,
```

Comment:
Can you imagine a case where these last two parameters are not the opposite of each other? They could be eliminated entirely (default to `true, false`):
```java
anySatisfy  = this.shortCircuitKeyValue(predicate);
allSatisfy  = !this.shortCircuitKeyValue(Predicates.not(predicate));
noneSatisfy = !this.shortCircuitKeyValue(predicate);
```

Reply:
looking at the above, it strikes me that this new function is just `anySatisfy` :smile:

Reply:
This came out pretty similar to UnifiedMap.shortCircuit() and UnifiedSet.shortCircuit() but with different parameter names. Was that on purpose? Any advice that applies here (like not needing two redundant parameters) probably also applies there.

Reply:
> This came out pretty similar to UnifiedMap.shortCircuit() and UnifiedSet.shortCircuit() but with different parameter names. Was that on purpose? Any advice that applies here (like not needing two redundant parameters) probably also applies there.

I fiddled with the parameter names, so yes it was on purpose.